xtuner/v1/datasets/utils.py

```diff
@@ -160,3 +161,31 @@ def replace_image_context_and_collect_media_data(
                             c["text"] = _c.replace("<IMG_CONTEXT>", "")
 
     return image_paths, video_paths
+
+
+def dict_to_sorted_string(input_dict):
```

Comment:
```python
dict1 = {"data": [{"b": 2, "a": 1}]}
dict2 = {"data": [{"a": 1, "b": 2}]}
```

Should `dict_to_sorted_string` return the same results? Besides, for this general conversion function, we need to add typehint to it and add some exception catching, e.g, if tuple is found in `input_dict`, an error should be raised,